.github/workflows/publish.yml

```diff
@@ -40,15 +40,33 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.NPM_TOKEN }}
 
+      - uses: actions/cache/restore@v3
+        id: cache
+        with:
+          path: dist
+          key: cache-${{ github.run_id }}-${{ github.run_attempt }}
+
       - name: prepare node
+        if: steps.cache.outputs.cache-hit != 'true'
         uses: knallbumm/kb-github-actions/.github/actions/prepare-node@pnpm
         with:
           npm_token: ${{ secrets.NPM_TOKEN }}
           checkout_modules: ${{ inputs.checkout_modules }}
           npm_registry: ${{ inputs.npm_registry }}
+          run_install: ${{ steps.cache.outputs.cache-hit != 'true' }}
 
-      - run: pnpm run build
+      - name: build
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: pnpm run build
+
+      - name: setup node
+        if: steps.cache.outputs.cache-hit == 'true'
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          registry-url: ${{ inputs.npm_registry }}
 
-      - run: pnpm publish --access restricted
+      - name: publish
+        run: npm publish --access restricted
```

Comment:
Why npm instead of pnpm

Reply:
Not sure whether the pnpm install actually takes a relevant amount of time, but I just went without it